db/compaction/compaction_outputs.cc

```diff
@@ -532,6 +551,31 @@ Status CompactionOutputs::AddRangeDels(
           smallest_candidate =
               InternalKey(*lower_bound, tombstone.seq_, kTypeRangeDeletion);
         }
+      } else if (lower_bound_from_range_tombstone) {
+        // There are two cases depending on whether the lower bound is a
+        // truncated range tombstone start key. If it is not truncated, then the
```

Comment:
Yes updated comments. If a range tombstone is not truncated when read from input file, it would have seqno = kMaxSequenceNumber and op_type = kTypeRangeDeletion. The previous compaction output file would receive this key as parameter `next_table_min_key` for `AddRangeDels()` and set the file's largest key to be no larger than it. This should prevent the range tombstone from being added to the previous file. But even if the range tombstone was added, correctness is preserved as the range tombstone will be truncated by file's largest key (which is before the range tombstone's start key).